### DISCUSSION TITLE
Semantic version is not followed

### DISCUSSION THREAD

**Discussion by artemptushkin** [aws/aws-sdk-java#2944] [Q&A]
I have this version update for `com.amazonaws:aws-java-sdk-bom` 

![image](https://user-images.githubusercontent.com/22632802/224368615-6dcdafd4-ed39-429d-83a8-0daa44c6730a.png)

It causes a compilation issue

```
error: incompatible types: AmazonDynamoDB cannot be converted to DynamoDbClient
    this.client = new AmazonDynamoDBLockClient(AmazonDynamoDBLockClientOptions.builder(builder.amazonDynamoDB, builder.tableName)
```

It's something not expected from a patch version update

**Comment by debora-ito** [comment]
`DynamoDbClient` is a Java SDK 2.x class - https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/dynamodb/DynamoDbClient.html

What probably happened is `AmazonDynamoDBLockClient` upgraded Java SDK to 2.x, that's where the "incompatible types" error is coming from. 

We are not the owners of `AmazonDynamoDBLockClient`, you can check for more info in their repo - https://github.com/awslabs/amazon-dynamodb-lock-client

**Reply by artemptushkin** [reply]
It's about bom, bom exposes a version of a library incompatible
